lib/iris/fileformats/netcdf.py

```diff
@@ -1994,77 +1885,97 @@ def _create_cf_coord_variable(self, cube, dimension_names, coord):
+                    raise ValueError(msg)
 
-            if coord in cf_coordinates:
+            if is_dimcoord:
                 # By definition of a CF-netCDF coordinate variable this
                 # coordinate must be 1-D and the name of the CF-netCDF variable
                 # must be the same as its dimension name.
-                cf_name = cf_dimensions[0]
-
-            # Get the values in a form which is valid for the file format.
-            points = self._ensure_valid_dtype(
-                coord.points, "coordinate", coord
-            )
+                cf_name = element_dims[0]
 
             # Create the CF-netCDF variable.
             cf_var = self._dataset.createVariable(
-                cf_name, points.dtype.newbyteorder("="), cf_dimensions
+                cf_name, data.dtype.newbyteorder("="), element_dims
             )
 
             # Add the axis attribute for spatio-temporal CF-netCDF coordinates.
-            if coord in cf_coordinates:
-                axis = iris.util.guess_coord_axis(coord)
+            if is_dimcoord:
+                axis = iris.util.guess_coord_axis(element)
                 if axis is not None and axis.lower() in SPATIO_TEMPORAL_AXES:
                     _setncattr(cf_var, "axis", axis.upper())
 
             # Add the data to the CF-netCDF variable.
-            cf_var[:] = points
+            cf_var[:] = data  # TODO: support dask streaming
 
-            # Create the associated CF-netCDF bounds variable.
-            self._create_cf_bounds(coord, cf_var, cf_name)
+            # Create the associated CF-netCDF bounds variable, if any.
+            self._create_cf_bounds(element, cf_var, cf_name)
+
+        # Add the data to the CF-netCDF variable.
+        cf_var[:] = data
 
         # Deal with CF-netCDF units and standard name.
-        standard_name, long_name, units = self._cf_coord_identity(coord)
+        if isinstance(element, iris.coords.Coord):
+            # Fix "degree" units if needed.
+            # TODO: rewrite the handler routine to a more sensible API.
+            _, _, units_str = self._cf_coord_identity(element)
+        else:
+            units_str = str(element.units)
 
-        if cf_units.as_unit(units).is_udunits():
-            _setncattr(cf_var, "units", units)
+        if cf_units.as_unit(units_str).is_udunits():
+            _setncattr(cf_var, "units", units_str)
 
+        standard_name = element.standard_name
         if standard_name is not None:
             _setncattr(cf_var, "standard_name", standard_name)
 
+        long_name = element.long_name
         if long_name is not None:
             _setncattr(cf_var, "long_name", long_name)
 
         # Add the CF-netCDF calendar attribute.
-        if coord.units.calendar:
-            _setncattr(cf_var, "calendar", coord.units.calendar)
+        if element.units.calendar:
+            _setncattr(cf_var, "calendar", str(element.units.calendar))
```

Comment:
@pp-mo I *think* a `calendar` attribute is only applicable to a `time` coordinate, and not a cell-measure or ancillary variable, see https://cfconventions.org/cf-conventions/cf-conventions.html#calendar

Reply:
True, but I thought in that case the unit simply wouldn't ever have a calendar ?